connectivity/check/deployment.go

```diff
@@ -413,6 +556,31 @@ func (ct *ConnectivityTest) validateDeployment(ctx context.Context) error {
 		}
 	}
 
+	perfPods, err := ct.client.ListPods(ctx, ct.params.TestNamespace, metav1.ListOptions{LabelSelector: "kind=" + kindPerfName})
+	if err != nil {
+		return fmt.Errorf("unable to list perf pods: %s", err)
+	}
+	for _, perfPod := range perfPods.Items {
+		ctx, cancel := context.WithTimeout(ctx, ct.params.ciliumEndpointTimeout())
+		defer cancel()
+		if err := ct.waitForCiliumEndpoint(ctx, ct.clients.src, ct.params.TestNamespace, perfPod.Name); err != nil {
+			return err
+		}
+		_, err := perfPod.GetLabels()["server"]
+		if err {
```

Comment:
Nit: In Go `err` is usually a var of type `error`. How about naming this `hasServer` or similar?

```suggestion
		_, hasServer := perfPod.GetLabels()["server"]
		if hasServer {
```